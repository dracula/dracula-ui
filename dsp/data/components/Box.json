{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-14T00:08:30.329Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Dracula UI standard spacing properties.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Dracula UI standard spacing properties.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
