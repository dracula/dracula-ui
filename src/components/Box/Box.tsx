import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
