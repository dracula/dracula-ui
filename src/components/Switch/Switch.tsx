import { ColorMap } from '../../base/colors'
import React, { HTMLAttributes } from 'react'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'
import classNamesDedupe from 'classnames/dedupe'

export const switchColors: Partial<ColorMap> = {
  white: 'drac-switch-white',
  cyan: 'drac-switch-cyan',
  green: 'drac-switch-green',
  orange: 'drac-switch-orange',
  pink: 'drac-switch-pink',
  purple: 'drac-switch-purple',
  red: 'drac-switch-red',
  yellow: 'drac-switch-yellow'
}

/** Switch Props. */
export interface SwitchProps
  extends HTMLAttributes<HTMLInputElement>,
    PaddingMixin,
    MarginMixin {
  /** The name of the form element. */
  name?: string

  /**
   * The Dracula UI theme color to be applied to the Switch.
   */
  color: keyof typeof switchColors

  /**
   * A disabled switch is unusable and un-clickable.
   */
  disabled?: boolean
}

/**
 * Switch is an abstraction pattern that represents toggles.
 *
 * Use the Switch component as an alternative for Checkboxes or Radios
 * when options are binary.
 *
 * Switches are implemented as Checkboxes under the hood,
 * and work normally with forms.
 */
export const Switch = React.forwardRef<HTMLInputElement, SwitchProps>(
  (props, ref) => {
    const { name, color, disabled, ...htmlProps } = props

    const finalProps = {
      name,
      disabled,
      ...htmlProps,
      className: classNamesDedupe(
        `drac-switch`,
        'drac-checkbox',
        props.className,
        switchColors[color],
        ...paddingMixin(props),
        ...marginMixin(props)
      )
    }

    return <input type="checkbox" ref={ref} {...cleanProps(finalProps)} />
  }
)

Switch.displayName = 'Switch'
