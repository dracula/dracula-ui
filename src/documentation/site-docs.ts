import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
