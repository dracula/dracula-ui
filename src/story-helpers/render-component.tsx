import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
