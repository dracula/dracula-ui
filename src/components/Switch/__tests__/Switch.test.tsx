import React from 'react'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '@/documentation/render-component'
import { Box } from '@/components/Box/Box'
import { Switch, SwitchProps, switchColors } from '@/components/Switch/Switch'

docs(Switch, {
  basic() {
    return snapshot(
      'Usage',
      Usage,
      'Render a toggle component using checkbox under the hood.'
    )
  },
  variations() {
    return [
      snapshot(
        'Colors',
        Colors,
        'Switches can be customized to use any of the Dracula UI theme colors.'
      ),
      snapshot(
        'Disabled',
        Disabled,
        'A disabled switch is unusable and un-clickable.'
      ),
    ]
  }
})

function Usage() {
  return (
    <Box>
      <Switch color="purple" id="demo" name="demo" defaultChecked />
      <label htmlFor="demo" className="drac-text drac-text-white">
        Checked
      </label>
    </Box>
  )
}

function Disabled() {
  return (
    <Box>
      <Box mb="xs">
        <Switch
          id="one"
          disabled
          defaultChecked
          color="white"
          name="one"
        />
        <label htmlFor="one" className="drac-text drac-text-white">
          Checked
        </label>
      </Box>
      <Box>
        <Switch
          id="two"
          disabled
          color="white"
          name="two"
        />
        <label htmlFor="two" className="drac-text drac-text-white">
          Unchecked
        </label>
      </Box>
    </Box>
  )
}

function Colors() {
  return (
    <Box style={{ display: 'flex', flexDirection: 'column' }}>
      {Object.keys(switchColors).map((color) => {
        return (
          <Box key={color} mb="xs">
            <Switch
              id={color}
              defaultChecked
              color={color as SwitchProps['color']}
              name="demo"
            />
            <label htmlFor={color} className="drac-text drac-text-white">
              Switch
            </label>
          </Box>
        )
      })}
    </Box>
  )
}
