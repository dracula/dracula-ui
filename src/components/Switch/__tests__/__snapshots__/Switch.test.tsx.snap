// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Switch Basic Usage 1`] = `
Object {
  "docs": "Render a toggle component using checkbox under the hood.",
  "html": "<div class=\\"drac-box\\">
  <input
    type=\\"checkbox\\"
    name=\\"demo\\"
    id=\\"demo\\"
    class=\\"drac-switch drac-checkbox drac-switch-purple\\"
    checked=\\"\\"
  /><label for=\\"demo\\" class=\\"drac-text drac-text-white\\">Checked</label>
</div>
",
  "react": "<Box>
  <Switch color=\\"purple\\" id=\\"demo\\" name=\\"demo\\" defaultChecked={true} />
  <label htmlFor=\\"demo\\" className=\\"drac-text drac-text-white\\">
    Checked
  </label>
</Box>;
",
  "title": "Usage",
}
`;

exports[`Site: Switch Colors 1`] = `
Object {
  "docs": "Switches can be customized to use any of the Dracula UI theme colors.",
  "html": "<div style=\\"display: flex; flex-direction: column\\" class=\\"drac-box\\">
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"white\\"
      class=\\"drac-switch drac-checkbox drac-switch-white\\"
      checked=\\"\\"
    /><label for=\\"white\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"cyan\\"
      class=\\"drac-switch drac-checkbox drac-switch-cyan\\"
      checked=\\"\\"
    /><label for=\\"cyan\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"green\\"
      class=\\"drac-switch drac-checkbox drac-switch-green\\"
      checked=\\"\\"
    /><label for=\\"green\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"orange\\"
      class=\\"drac-switch drac-checkbox drac-switch-orange\\"
      checked=\\"\\"
    /><label for=\\"orange\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"pink\\"
      class=\\"drac-switch drac-checkbox drac-switch-pink\\"
      checked=\\"\\"
    /><label for=\\"pink\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"purple\\"
      class=\\"drac-switch drac-checkbox drac-switch-purple\\"
      checked=\\"\\"
    /><label for=\\"purple\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"red\\"
      class=\\"drac-switch drac-checkbox drac-switch-red\\"
      checked=\\"\\"
    /><label for=\\"red\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"yellow\\"
      class=\\"drac-switch drac-checkbox drac-switch-yellow\\"
      checked=\\"\\"
    /><label for=\\"yellow\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
</div>
",
  "react": "<Box style={{ display: \\"flex\\", flexDirection: \\"column\\" }}>
  <Box mb=\\"xs\\">
    <Switch id=\\"white\\" defaultChecked={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"white\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"cyan\\" defaultChecked={true} color=\\"cyan\\" name=\\"demo\\" />
    <label htmlFor=\\"cyan\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"green\\" defaultChecked={true} color=\\"green\\" name=\\"demo\\" />
    <label htmlFor=\\"green\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"orange\\" defaultChecked={true} color=\\"orange\\" name=\\"demo\\" />
    <label htmlFor=\\"orange\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"pink\\" defaultChecked={true} color=\\"pink\\" name=\\"demo\\" />
    <label htmlFor=\\"pink\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"purple\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"purple\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"red\\" defaultChecked={true} color=\\"red\\" name=\\"demo\\" />
    <label htmlFor=\\"red\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
  <Box mb=\\"xs\\">
    <Switch id=\\"yellow\\" defaultChecked={true} color=\\"yellow\\" name=\\"demo\\" />
    <label htmlFor=\\"yellow\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </Box>
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Switch Disabled 1`] = `
Object {
  "docs": "A disabled switch is unusable and un-clickable.",
  "html": "<div class=\\"drac-box\\">
  <div class=\\"drac-box drac-mb-xs\\">
    <input
      type=\\"checkbox\\"
      name=\\"one\\"
      disabled=\\"\\"
      id=\\"one\\"
      class=\\"drac-switch drac-checkbox drac-switch-white\\"
      checked=\\"\\"
    /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
  </div>
  <div class=\\"drac-box\\">
    <input
      type=\\"checkbox\\"
      name=\\"two\\"
      disabled=\\"\\"
      id=\\"two\\"
      class=\\"drac-switch drac-checkbox drac-switch-white\\"
    /><label for=\\"two\\" class=\\"drac-text drac-text-white\\">Unchecked</label>
  </div>
</div>
",
  "react": "<Box>
  <Box mb=\\"xs\\">
    <Switch
      id=\\"one\\"
      disabled={true}
      defaultChecked={true}
      color=\\"white\\"
      name=\\"one\\"
    />
    <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
      Checked
    </label>
  </Box>
  <Box>
    <Switch id=\\"two\\" disabled={true} color=\\"white\\" name=\\"two\\" />
    <label htmlFor=\\"two\\" className=\\"drac-text drac-text-white\\">
      Unchecked
    </label>
  </Box>
</Box>;
",
  "title": "Disabled",
}
`;
